### ISSUE TITLE
Feature request: pass ref to useBody args callback

### ISSUE THREAD

**Issue by RobRendell** [react-spring/use-cannon#53]
Sorry to raise another feature request - and I'm not even sure if this is a valid idea.

`use-cannon` has been awesome, and my rolling die react component is working really well.  The current class looks something like

```
function Die(props) {
   const geometry = useMemo(() => (buildGeometry(props.dieType)), [props.dieType]);
   const [ref] = useConvexPolyhedron(() => ({ args: geometry }));
   return (<mesh geometry={geometry} ref={ref} />);
}
```

Sometimes however I want to render the dice without physics, so I figured I might have one component which is just the mesh, and wrap it with another component which has the physics.  Something like

```
function DieObject(props) {
   const geometry = useMemo(() => (buildGeometry(props.dieType)), [props.dieType]);
   return (<mesh geometry={geometry} ref={props.physicsRef} />);
}

function Die(props) {
   const [ref] = useConvexPolyhedron(() => ({ args: /* TODO */ }));
   return (<DieObject physicsRef={ref}  {...props} />);
}
```
As you can see, I don't have the geometry of the mesh available to initialise the body.

However, it occurs to me that the ref function gets the the mesh object, which has access to the geometry.  If the ref was passed as a parameter to the args callback (after it has been initialised), then the Die component could be

```
function Die(props) {
   const [ref] = useConvexPolyhedron((ref) => ({ args: ref.current.geometry }));
   return (<DieObject physicsRef={ref} {...props} />);
}
```

Does this seem like it would be possible?  Is it a good idea?

**Comment by drcmda** [comment]
i think you can access the ref, it's part of the local scope:

```jsx
function Die(props) {
   const [ref] = useConvexPolyhedron(() => ({ args: ref.current.geometry }));
```

**Comment by RobRendell** [comment]
I initially tried to use the external ref as you suggest, but when I do that I get a run-time error:

`Uncaught ReferenceError: Cannot access 'ref' before initialization`

**Comment by drcmda** [comment]
it shouldn't be, but yeah, that's because of forwardref which was added recently, it worked before that  :-(

```jsx
function useBody(type: BodyShapeType, fn: BodyFn, argFn: ArgFn, deps: any[] = []): Api {
  const { ref: fwdRef } = fn(0)
```

**Comment by codynova** [comment]
Resolved in #54, closing
